### PR TITLE
[BIOMAGE-2020] Deploy to HMS

### PR DESCRIPTION
**Pull request by ivababukova**
# Background
This PR was used to deploy Cellenics to the HMS account. The changes here are fixing small issues found during the Cellenics deployment and are introducing the new environment into the Github actions flow.

#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-2020

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes
- Removed the notion of read-only cluster users (file infra/cluster_users). It is not used.
- Adapted the workflows to execute for both Biomage and HMS environment names, where for HMS we can only run with production environment type.
- Changed the file structure of the cluster admins files to allow for different rights, based on environment-name.
- Changed the git.url when installing flux to equal to `https://` for cases when flux has to be installed in read-only mode. This is done to fix a workflow error.

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock  (with latest production data downloaded with biomage experiment pull)
- [x] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/hms-dbmi-cellenics/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/hms-dbmi-cellenics/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/hms-dbmi-cellenics/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: .github/workflows/deploy-changed-cf.yaml [hunk 1]
```diff
@@ -150,13 +150,16 @@ jobs:
...
+      deploy-rds: ${{ steps.set-name.outputs.deploy-rds }}
     strategy:
       max-parallel: 1
       matrix:
         environment-type: ${{ fromJson(needs.setup.outputs.matrix) }}
-        environment-name: [Biomage]
+        environment-name: [Biomage, HMS]
         template: ${{fromJson(needs.get-modified-templates.outputs.files)}}
+        exclude:
+          - environment-name: HMS
```
Comment (kafkasl) [inline .github/workflows/deploy-changed-cf.yaml:161]: so what's this? you are adding HMS to environment and then excluding it?

File: .github/workflows/deploy-infra.yaml [hunk 2]
```diff
@@ -357,10 +363,20 @@ jobs:
           helm repo add fluxcd https://charts.fluxcd.io
           kubectl apply -f https://raw.githubusercontent.com/fluxcd/helm-operator/master/deploy/crds.yaml
 
-          [[ $ENVIRONMENT_NAME = "Biomage" ]] && FLUX_GIT_READ_ONLY=false || FLUX_GIT_READ_ONLY=true
+          FLUX_GIT_READ_ONLY=false
+          GIT_URL=git@github.com:${{ env.RELEASES_REPOSITORY }}
+
+          if [[ $ENVIRONMENT_NAME = "HMS" ]]
```
Comment (kafkasl) [inline .github/workflows/deploy-infra.yaml]: if we are going to do this for flare, then it's probably to have `if [[ $ENVIRONMENT_NAME != "Biomage" ]]` ?

**Review by kafkasl — APPROVED** [review]

File: .github/workflows/deploy-infra.yaml (hunk 2 above)
Comment (ivababukova) [reply .github/workflows/deploy-infra.yaml]: good point, I will flip the logic of the if statement

Comment (ivababukova) [reply .github/workflows/deploy-infra.yaml]: done

**Pull request merged**